kiwi/storage/subformat/vagrant_base.py

```diff
@@ -0,0 +1,266 @@
+      ``Vagrant.configure("2") do |config|`` and ``end``. This class will
+      provide it with the correct base indentation.
+    """
+    def post_init(self, custom_args):
+        """
+        vagrant disk format post initialization method
+
+        store vagrantconfig information provided via custom_args
+
+        :param dict custom_args:
+            Contains instance of xml_parse::vagrantconfig
+
+            .. code:: python
+
+                {'vagrantconfig': object}
+        """
+        if not custom_args or 'vagrantconfig' not in custom_args:
+            raise KiwiFormatSetupError(
+                'object init requires custom_args hash with a vagrantconfig'
+            )
+
+        if not custom_args['vagrantconfig']:
+            raise KiwiFormatSetupError(
+                'no vagrantconfig provided'
+            )
+
+        self.image_format = 'vagrant.' + self.provider + '.box'
+        self.vagrantconfig = custom_args['vagrantconfig']
+
+    def create_image_format(self):
+        # type: () -> None
+        """
+        Create a vagrant box for any provider. This includes:
+
+        * creation of box metadata.json
+        * creation of box Vagrantfile
+        * creation of result format tarball from the files created above
+        """
+        self.temp_image_dir = mkdtemp(prefix='kiwi_vagrant_box.')
+
+        box_img_files = self.create_box_img(self.temp_image_dir)
+
+        metadata_json = os.sep.join([self.temp_image_dir, 'metadata.json'])
+        with open(metadata_json, 'w') as meta:
+            meta.write(self._create_box_metadata())
+
+        vagrantfile = os.sep.join([self.temp_image_dir, 'Vagrantfile'])
+        with open(vagrantfile, 'w') as vagrant:
+            vagrant.write(self._create_box_vagrantconfig())
+
+        Command.run(
+            [
+                'tar', '-C', self.temp_image_dir,
+                '-czf', self.get_target_file_path_for_format(self.image_format),
+                os.path.basename(metadata_json),
+                os.path.basename(vagrantfile)
+            ] + [
+                os.path.basename(box_img_file)
+                for box_img_file in box_img_files
+            ]
+        )
+
+    def store_to_result(self, result):
+        # type: (Result) -> None
+        """
+        Store result file of the vagrant format conversion into the
+        provided result instance. In this case compression is unwanted
+        because the box is already created as a compressed tarball
+
+        :param object result: Instance of Result
+        """
+        result.add(
+            key='disk_format_image',
+            filename=self.get_target_file_path_for_format(
+                self.image_format
+            ),
+            use_for_bundle=True,
+            compress=False,
+            shasum=True
+        )
+
+    @classmethod
+    def get_additional_metadata(cls):
+        # type: () -> Dict[str, str]
+        return {}
```

Comment:
Could we avoid that type of comments: In this particular method it's also wrong. it returns an empty dict
methods of the public interface should have a doc string as explained before. private methods (_foo) doesn't need it and could take some information if not clear by the code. I usually avoid any comment block that is of no interest because not in the public api and try to write the code to be as readable as possible. From my experience if there are comment blocks that looks like code either the comment or the code is wrong :) So let's focus on the minimal needed and be as explicit and clear in the code as possible. Hope that makes sense to you too

Reply:
I'll drop them.